### PR TITLE
Update z3 tests

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

**Comment by ekpyron** [comment]
It's both funny and annoying how predictably these things get inconsistent behaviour in osx.

**Comment by leonardoalt** [comment]
> It's both funny and annoying how predictably these things get inconsistent behaviour in osx.

haha yea exactly

**Comment by leonardoalt** [comment]
STACK TOO DEEP

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,24 +9,24 @@ version: 2.1
...
-    # solbuildpackpusher/solidity-buildpack-deps:ubuntu1604.clang.ossfuzz-22
-    default: "solbuildpackpusher/solidity-buildpack-deps@sha256:baf46ad78b131eb6b0bf98d50e9330437a338c6893b53af3853b36f6f4ab18ae"
+    # solbuildpackpusher/solidity-buildpack-deps:ubuntu1604.clang.ossfuzz-23
+    default: "solbuildpackpusher/solidity-buildpack-deps@sha256:7a84f67829ec2fdcfc8a61ebf22f9dc27b04a9edeb1531565ab3e0ebb1cbc4a0"
   emscripten-docker-image:
     type: string
-    # solbuildpackpusher/solidity-buildpack-deps:emscripten-14
-    default: "solbuildpackpusher/solidity-buildpack-deps@sha256:d7d4ec28cdc3e61fc6a048b88ebdff1c5e67f3353cf1739cbaea07ec2481eb16"
+    # solbuildpackpusher/solidity-buildpack-deps:emscripten-15
+    default: "solbuildpackpusher/solidity-buildpack-deps@sha256:f86cb1fb4631b1fbb927149f60caea7b095be5b85d379456eee09cb105e7f225"
```
Comment (ekpyron) [inline .circleci/config.yml]:
```suggestion
    default: "solbuildpackpusher/solidity-buildpack-deps@sha256:e2729bd6c0734bb49d04e2688fec3dcf880394f351f69358dd2bd92c0f6fd309"
```
given https://github.com/ethereum/solidity/pull/14074#issuecomment-1486833102

**Review by ekpyron — APPROVED** [review]
The ossfuzz hash still seems to be the one from the previous build in https://github.com/ethereum/solidity/pull/14074 - but I'm fine with ignoring that.

**Pull request merged**